fix: Fixed the hiding behavior of sidebar with footer in Documentation.vue

sahitya-chandra opened the pull request:
<!--- Provide a general summary of your changes in the Title above. -->
<!--- Do not include backticks (`) in the Title above.  -->

## Description

<!--- Describe your changes in detail. -->
<!--- Why is this change required? What problem does it solve? -->
<!--- Do not include backticks (`).  -->
1. Fixed the hiding of the sidebar with the footer when we scroll to the end of the Documentation page.
2. Removed the 'QDrawer' for better and controlled behavior of the Sidebar and made it hidden on the small screen.
3. Structured CSS in proper order.

## Related issue

<!--- Replace only the '000' with the issue number. -->
<!--- Do not include a URL. -->

fixed #908 

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, the tests you ran to. -->
<!--- see how your change affects other areas of the code, etc. -->
<!--- Do not include backticks (`).  -->
I have tested all the changes locally.

## Screenshots (if appropriate):

[Screencast from 2025-02-17 15-21-42.webm](https://github.com/user-attachments/assets/cdc9a756-d809-4c3e-90c7-ac391fb19c63)

[Screencast from 2025-02-17 16-14-06.webm](https://github.com/user-attachments/assets/4f33fab4-2249-461a-a2f4-b4df4d49165c)

## Types of changes

<!--- What types of changes does your code introduce? Put an 'x' in all the boxes that apply: -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an 'x' in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

sahitya-chandra commented:
@dpgiakatos Sir please review this code...

dpgiakatos reviewed (changes requested):
This PR deletes the QDrawer and re-implements it. However, the contributor should have used the existing QDrawer and fixed the issue instead of deleting and re-implementing it. We don't need to reinvent the wheel, as this approach will take too much time for maintenance. Thus, this PR needs a lot of changes and should continue to use Quasar components.

sahitya-chandra commented:
@dpgiakatos Sir Due to the implicit behavior of QDrawer it has infinite height because of this I have re-implemented the sidebar for static behavior.
Can I use any other Quasar component other than QDrawer if need be?

dpgiakatos commented:
Yes, you can use other quasar components.

sahitya-chandra commented:
@dpgiakatos Sir I have added the QSplitter component and organized the CSS in proper order. Removed multiple media-query.
Please review the code and let me know if there is any scope for correction.

[Screencast from 2025-02-18 11-15-30.webm](https://github.com/user-attachments/assets/7a0cb4bf-4576-4259-a91f-d5a1bc7b0994)

sahitya-chandra commented:
Also Sir, I have not deleted any code just organised it.

dpgiakatos commented on src/views/Documentation.vue:
```diff
@@ -135,72 +134,74 @@ const sectionActive = ref('')
-      <div v-for="(sec, idx) in sections" :key="idx">
-        <div class="text-weight-light">
-          {{ $t(`documentationPage.sectionsTitle.${sec.sectionsTitle}`) }}
+  <QSplitter>
```
You haven't configured the component. You have to ad the `v-model`.
Also, add `disable` property.
Read the documentation:
https://quasar.dev/vue-components/splitter

sahitya-chandra replied:
@dpgiakatos Sir Where should I add `disable` property...

sahitya-chandra replied:
@dpgiakatos Sir as you mentioned above to achieve old version behavior. 
 I have removed the QSplitter component because it was causing problems with anchor navigation and 
 I have added QCard and QCardSection for proper navigation and better scrolling of Sidebar.

dpgiakatos reviewed (changes requested):
Address my inline comment. Additionally, I would like the page to have the same behavior as the old version. Previously, when scrolling to a section, the contents would be visible on the right. In this version, you have to scroll to the top to see them. I would like to restore the same behavior as the old page, where the contents are visible on the right when scrolling to a section.

sahitya-chandra commented:
This the current behavior. 

[Screencast from 2025-02-18 18-54-19.webm](https://github.com/user-attachments/assets/c1e11727-ddea-47d9-b08a-abe6dadc3591)

sahitya-chandra commented:
Sir I have also removed multiple media queries to reduce redundancy in the code.

dpgiakatos reviewed (approved):
Thanks

sahitya-chandra commented:
Thanks for approving my changes Sir...🚀
